core/src/main/scala/io/snappydata/SnappyDaemons.scala

```diff
@@ -131,7 +128,7 @@ object SnappyAnalyticsService extends Logging {
 	}
 
 	def getUIInfo: Seq[UIAnalytics] = {
-		val currentTableStats = tableStats.get()
+		val currentTableStats = tableStats.get().clone()
```

Comment:
The getters and updates are two different threads right? In that case this the copy is not thread-safe and better to just use a ConcurrentMap instead of a map (remove the AtomicReference too). In scala you can either use TrieMap, or java.util.concurrent.ConcurrentHashMap.toScala. Both of these will return scala Maps so that can be passed around freely without worrying about making a copy.

The call to getTableSize in this class also suffers from thread safety issue. Using a concurrent map will correct that too.